### PR TITLE
fix: Status codes minor issues

### PR DESCRIPTION
**Pull request by dantetam**
### Summary

Fixes [AMP-37687](https://amplitude.atlassian.net/browse/AMP-37687)

### Checklist

* [ X ] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-iOS/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  No

### CODE REVIEW

File: src/main/java/com/amplitude/HttpCall.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ protected Response syncHttpCallWithEventsBuffer(List<Event> events)
       // Also SocketTimeoutException, when the HTTP request times out.
       JSONObject timesOutResponse = new JSONObject();
       timesOutResponse.put("status", Status.TIMEOUT);
```
Comment (yuhao900914) [inline src/main/java/com/amplitude/HttpCall.java:89]: Should this be `Status.UNKNOWN` ?

Comment (dantetam) [reply src/main/java/com/amplitude/HttpCall.java:89]: I've decided to use the TIMEOUT status for all of these errors/cases. In the documentation, it says that TIMEOUT is used for any server side or client side HTTP request that times out. This is the best way to organize it I think, let me know if you had more context or ideas

Comment (yuhao900914) [reply src/main/java/com/amplitude/HttpCall.java:89]: Yea, but I looked at the ticket in Jira, it says the test case will return UNKNOW in the response.

Comment (dantetam) [reply src/main/java/com/amplitude/HttpCall.java:89]: Yeah, that's why the number was changed to 408. So it should point to TIMEOUT

Comment (yuhao900914) [reply src/main/java/com/amplitude/HttpCall.java:89]: Oh, I see.

**Review by yuhao900914 — APPROVED** [review]

**Review by qingzhuozhen — APPROVED** [review]

**Pull request merged**